test/instance_unittest.cc

```diff
@@ -0,0 +1,77 @@
+#include "../src/configuration.h"
+#include "../src/environment.h"
+#include "../src/instance.h"
+#include "../src/resource.h"
+#include "../src/updater.h"
+#include "gtest/gtest.h"
+
+namespace google {
+
+namespace {
+
+class InstanceTest : public ::testing::Test {
+ protected:
+  const MetadataStore::Metadata&& GetResourceMetadata(
+      const MetadataUpdater::ResourceMetadata* rm) {
+    return std::move(rm->metadata);
+  }
+
+  MonitoredResource GetMonitoredResource(
+      const MetadataUpdater::ResourceMetadata* rm) {
+    return rm->resource;
+  }
+
+  std::vector<std::string> GetResourceIds(
+      const MetadataUpdater::ResourceMetadata* rm) {
+    return rm->ids;
+  }
+};
+
+TEST_F(InstanceTest, GetInstanceMonitoredResource) {
+  Configuration config(std::istringstream(
+      "InstanceResourceType: gce_instance\n"
+      "InstanceId: 1234567891011\n"
+      "InstanceZone: us-east1-b\n"
+  ));
+  Environment env(config);
+  MonitoredResource mr_actual =
+      InstanceReader::InstanceResource(env);
+  MonitoredResource mr_expected(
+      "gce_instance",
+      {{"instance_id", "1234567891011"}, {"zone", "us-east1-b"}});
+  EXPECT_EQ(mr_expected, mr_actual);
+}
+
+TEST_F(InstanceTest, GetInstanceMonitoredResourceWithEmptyConfig) {
+  Environment env(*(new Configuration()));
+  MonitoredResource mr_actual =
+      InstanceReader::InstanceResource(env);
+  MonitoredResource mr_expected(
+      "gce_instance", {{"instance_id", ""}, {"zone", ""}});
```

Comment:
I'm confused on this expectation.  Reading the code, I think this will actually make an HTTP call to the metadata server.  Is it possible to execute this test without mocks set up?

Reply:
+1. Totally missed that. Let's remove this test for now and add it when we have mocks.